rust/crd/src/lib.rs

```diff
@@ -691,6 +687,37 @@ fn cores_from_quantity(q: String) -> Result<String, Error> {
     Ok((cores as u32).to_string())
 }
 
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct VolumeMounts {
+    pub volume_mounts: Option<Vec<VolumeMount>>,
+}
+
+impl Atomic for VolumeMounts {}
+
+impl<'a> IntoIterator for &'a VolumeMounts {
+    type Item = &'a VolumeMount;
+    type IntoIter = slice::Iter<'a, VolumeMount>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.volume_mounts.as_deref().unwrap_or_default().iter()
+    }
+}
+
+impl From<VolumeMounts> for Vec<VolumeMount> {
+    fn from(value: VolumeMounts) -> Self {
+        value.volume_mounts.unwrap_or_default()
+    }
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NodeSelector {
+    pub node_selector: Option<BTreeMap<String, String>>,
+}
+
+impl Atomic for NodeSelector {}
```

Comment:
There is thus [NodeSelector](https://docs.rs/k8s-openapi/latest/k8s_openapi/api/core/v1/struct.NodeSelector.html) in k8s_openapi, but since it was not used before and im not sure about the `Atomic` im fine with leaving as is.

Reply:
`nodeSelector` is not used and apparently exists only for compatibility reasons. If I would have used `NodeSelector` from k8s_openapi then I would have restricted the possible values and perhaps introduced a breaking change. I wanted to avoid changing the CRD in this pull request (except fixing the non-camel-cased properties introduced in #217 which was not released yet) and therefore did not change this.